src/compiler/parser.ts

```diff
@@ -666,6 +668,9 @@ namespace ts {
             sourceFile.languageVariant = getLanguageVariant(sourceFile.fileName);

```

Comment:
switched the getLanguageVariant to use the scriptKind as now it will always be set.